### PR TITLE
creating a vesion command to get clusterversion

### PR DESCRIPTION
**Pull request by sferich888**
@juanvallejo  can you take a look and let me know what you think.

### CODE REVIEW

**Comment by juanvallejo** [comment]
cc @deads2k

File: pkg/cmd/version.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	if len(o.baseDir) == 0 {
+		return fmt.Errorf("--base-dir must not be empty")
+	}
+	return nil
+}
+
+func (o *VersionOptions) Run() error {
+	// next, ensure we're able to proceed writing data to specified destination
+	if err := o.ensureDirectoryViable(o.baseDir, o.overwrite); err != nil {
+		log.Printf("Failed to setup %q", o.baseDir)
```
Comment (juanvallejo) [inline pkg/cmd/version.go]: I think we can depend on the error message here. Logging seems redundant

File: pkg/cmd/version.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+			if err := o.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVar(&o.baseDir, "base-dir", "must-gather", "Root directory used for storing all gathered cluster operator data. Defaults to $(PWD)/must-gather")
+	cmd.Flags().BoolVar(&o.overwrite, "overwrite", false, "If true, allow this command to write to an existing location with previous data present")
```
Comment (juanvallejo) [inline pkg/cmd/version.go]: You can safely remove this flag. Also, default `o.overwrite` to true under `NewVersionOptions`

File: pkg/cmd/version.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+		return err
+	}
+
+	o.builder = resource.NewBuilder(o.configFlags)
+	return nil
+}
+
+func (o *VersionOptions) Validate() error {
+	if len(o.args) >= 1 {
+		return fmt.Errorf("this option takes not arguments")
```
Comment (juanvallejo) [inline pkg/cmd/version.go]: `len(o.args) > 0` reads a bit better imo. Also, how about rephrasing to `this command does not take arguments`

File: pkg/cmd/version.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+	clusterVersion, err := o.dynamicClient.Resource(configv1.SchemeGroupVersion.WithResource("clusterversions")).List(metav1.ListOptions{})
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	objToPrint := runtime.Object(clusterVersion)
+
+	filename := fmt.Sprintf("%s.yaml", "clusterversion")
+
+	if err := o.fileWriter.WriteFromResource(path.Join(o.baseDir, "/"+filename), objToPrint); err != nil {
```
Comment (juanvallejo) [inline pkg/cmd/version.go]: maybe consider putting these under `must-gather/resources/config.openshift.io/clusterversions.yaml`

Comment (juanvallejo) [reply pkg/cmd/version.go]: This is the current location where we place them with the `inspect` command.

**Comment by sferich888** [comment]
https://github.com/openshift/must-gather/pull/22 already collects this.

**Pull request closed without merging**